.github/workflows/workflow-build-push-container-github-registry.yml

```diff
@@ -0,0 +1,142 @@
+        run: |
+          gh run cancel ${{ github.run_id }}
+          gh run watch ${{ github.run_id }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Log in to the github container registry (GCR)
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+        with:
+          registry: ${{ inputs.registry }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Cache Docker layers
+        uses: actions/cache@v3
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: ${{ runner.os }}-buildx
+
+      - name: Extract branch name
+        shell: bash
+        run: echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_OUTPUT
+        id: extract_branch_name
+
+      - name: Build and push (3 tags | commit number, pr number and branch name) (unmerged)
+        id: docker_build_and_tag_unmerged
+        if: github.event.pull_request.merged == false
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          push: true
+          tags: |
+            ${{ inputs.registry }}/${{ inputs.container-name }}:${{ github.event.number }}
+            ${{ inputs.registry }}/${{ inputs.container-name }}:${{ inputs.tag }}
+            ${{ inputs.registry }}/${{ inputs.container-name }}:${{ steps.extract_branch_name.outputs.branch }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,mode=max,dest=/tmp/.buildx-cache-new
+
+      - name: Build and push (2 tags | commit number and branch name) (merged)
+        id: docker_build_and_tag_merged
+        if: github.event.pull_request.merged == true
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          push: true
+          tags: |
+            ${{ inputs.registry }}/${{ inputs.container-name }}:${{ inputs.tag }}
+            ${{ inputs.registry }}/${{ inputs.container-name }}:${{ github.event.pull_request.base.ref }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,mode=max,dest=/tmp/.buildx-cache-new
+
+      - name: Refresh Cache
+        run: |
+          rm -rf /tmp/.buildx-cache
+          mv /tmp/.buildx-cache-new /tmp/.buildx-cache
+
+      - name: Image digest (latest)
+        if: github.event.pull_request.merged == false
+        run: echo ${{ steps.docker_build_and_tag_unmerged.outputs.digest }}
+
+      - name: Image digest (latest)
+        if: github.event.pull_request.merged == true
+        run: echo ${{ steps.docker_build_and_tag_merged.outputs.digest }}
+
+  remove-old-image:
+    runs-on: ubuntu-latest
+    needs: build-push-image
+    steps:
+      - name: Checkout github.com/ai-cfia/devops
+        uses: actions/checkout@v2
```

Comment:
this should be necessary; here you're checking out the source, manually installing the dependencies when it should be a single line to install from the github repo:

https://docs.readthedocs.io/en/stable/guides/private-python-packages.html

Reply:
Do you mean "this shouldn't be necessary" ?

Reply:
Also, why replace something easily readable to the user like:

```bash
- name: Checkout github.com/ai-cfia/devops
        uses: actions/checkout@v2
        with:
          repository: 'ai-cfia/devops'
          ref: 26-as-a-devops-i-want-to-create-unit-tests-for-the-remove-previous-imagepy-script
          token: ${{ secrets.GITHUB_TOKEN }}
```

With a URL like this:
```bash
git+https://${{GITHUB_USER}}:${{GITHUB_PAT}}@github.com/ai-cfia/devops.git@26-as-a-devops-i-want-to-create-unit-tests-for-the-remove-previous-imagepy-script
```
I really don't see the utility of this change....

Reply:
Done!